### PR TITLE
Return actions, predictions, confidences and policies in rasa_core.evaluate

### LINKED ISSUE #1235: Include per-action policy information in evaluate.py

### PR DESCRIPTION
**Pull request by ricwo**
**Proposed changes**:
- fixes #1235 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_core/evaluate.py [hunk 1]
```diff
@@ -333,8 +333,9 @@ def _collect_user_uttered_predictions(event,
 def _collect_action_executed_predictions(processor, partial_tracker, event,
                                          fail_on_prediction_errors):
     action_executed_eval_store = EvaluationStore()
+    print("in here", partial_tracker.current_state())
```
Comment (tmbo) [inline rasa_core/evaluate.py]: found it :P

Comment (amn41) [reply rasa_core/evaluate.py]: @tmbo will you review this then?

File: rasa_core/evaluate.py [hunk 2]
```diff
@@ -403,14 +369,22 @@ def _predict_tracker_actions(tracker, agent, fail_on_prediction_errors=False,
...
         if isinstance(event, ActionExecuted):
-            action_executed_result, policy = \
+            action_executed_result, policy, confidence = \
                 _collect_action_executed_predictions(
                         processor, partial_tracker, event,
                         fail_on_prediction_errors
                 )
             tracker_eval_store.merge_store(action_executed_result)
+            tracker_actions.append(
+                    {"action": action_executed_result.action_targets[0],
```
Comment (tmbo) [inline rasa_core/evaluate.py]: can this never ever be empty?

File: rasa_core/evaluate.py [hunk 3]
```diff
@@ -420,7 +394,16 @@ def _predict_tracker_actions(tracker, agent, fail_on_prediction_errors=False,
         else:
             partial_tracker.update(event)
 
-    return tracker_eval_store, partial_tracker
+    return tracker_eval_store, partial_tracker, tracker_actions
+
+
+def _in_training_data_fraction(action_list):
```
Comment (tmbo) [inline rasa_core/evaluate.py]: comment please

File: rasa_core/evaluate.py [hunk 4]
```diff
@@ -463,16 +444,15 @@ def collect_story_predictions(
...
+    in_training_data_fraction = _in_training_data_fraction(action_list)
 
     log_evaluation_table([1] * len(completed_trackers),
                          "END-TO-END" if use_e2e else "CONVERSATION",
                          report, precision, f1, accuracy,
                          in_training_data_fraction,
                          include_report=False)
 
-    return story_eval_store, failed, in_training_data_fraction
+    return story_eval_store, failed, action_list, in_training_data_fraction
```
Comment (tmbo) [inline rasa_core/evaluate.py]: might make sense to make a namedtuple out of this? I think for items is quite a lot

File: rasa_core/evaluate.py (hunk 2 above)
Comment (ricwo) [reply rasa_core/evaluate.py]: I don't believe so, because `processor.predict_next_action()` always predicts _something_

**Pull request merged**